### ISSUE TITLE
Problem when max-instances for applet is changed.

### ISSUE THREAD

**Issue by pdcurtis** [linuxmint/Cinnamon#3127]
I am running Mint 16  and have been modifying an applet to allow multiple instances in the panel. An anomaly occurs when the number of instances is increased from or decreased to 1. When  metadata.json has the line  "max-instances": "1", or the line is missing the copy providing persistence of settings-schema.json in ~/.cinnamon/config/applet@name/ has a filename like  applet@name.json and if max-instances is higher it has files with a format nnn.json. If you change max-instances from 1 to 2 instances in metadata.json and add an extra instance you get three files and three tabs in 'settings' but only two instances in the panel. This also occurs in cinnamon 2.2 running over trusty - sorry I did not spot it earlier.

This can obviously be sorted by total removal and reinstalling or hand crafting but there is an associated problem that there is no easy way to let users know the my next  update planned which will allow multiple instances will screw them up until there is a reinstall and even then they will lose their settings and some cumulative data. 

In the longer term I suggest a mechanism to  warn users starting such updates from Settings -> Applets -> Get More Online Tab and push them to the web site where they can be alerted to problems.  This could also be appropriate where applets have dependencies or have unusual installation mechanisms such as scripts. This could be an addition to metadata.json which would allow backward compatibility.

**Comment by NikoKrause** [comment]
Is this still an issue in 3.2?

**Comment by pdcurtis** [comment]
@NikoKrause Yes it is still an issue in 3.2 but one which is both minor and unlikely to be fixed as it is rare but totally predictale and applet writers can warn people: see post by @mtwebster 

To recap: It is the result of changes in Cinnamon (2.2??) to allow multiple instances which still needed to support the legacy applets for earlier versions of Cinnamon. It only occurs if an applet is upgraded to allow multiple instances without a complete reinstall by a user. At the time several applets would have benefited from a fix but as long as people are aware a reinstall is needed it is not a problem and, but, far as I know, this issue is the only place other than my own applet documentation to make writers aware of the potential for a problem otherwise I would have closed it.

This brings up the issue of documentation for applet writers once more especially with the major change in 3.2 to allow vertical tool bars - I have yet to find documentation of what applet writers need to do (although to be fair I have not looked very hard yet!)

**Comment by icarter09** [comment]
@pdcurtis are you still experiencing this issue?

**Comment by pdcurtis** [comment]
I believe the issue is still present but will never be fixed in the system if you read above.
But this may still be the only place it is recorded!
A work round is a button in settings to do a tidy up of old files as I have done in another applet when changes to files providing 'persistence' were made.
